lgc/patch/PatchBufferOp.cpp

```diff
@@ -560,15 +556,12 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
     copyMetadata(rhsPtrToInt, rhs);
 
     Value *const difference = m_builder->CreateSub(lhsPtrToInt, rhsPtrToInt);
-    Constant *const size = ConstantExpr::getSizeOf(ty);
-    Value *const elementDifference = m_builder->CreateExactSDiv(difference, size);
 
     // Record the call instruction so we remember to delete it later.
-    m_replacementMap[&callInst] = std::make_pair(nullptr, nullptr);
+    m_replacementMap[ptrDiff] = std::make_pair(nullptr, nullptr);
 
-    callInst.replaceAllUsesWith(elementDifference);
-  } else
-    llvm_unreachable("Should never be called!");
```

Comment:
Why was the else removed?

Reply:
The old code made an assumption about the prefix of the names of the "intrinsics". The new code no longer does that.

By the way, this probably makes the new version slightly less efficient under the current llvm-dialects implementation using string comparisons. My intention is to implement the same metadata "opcode" trick that the builder recorder uses and make it switchable so that we can actually measure the performance impact.

Reply:
Makes sense. Good luck measuring the performance impact ;)